standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2736,15 +2736,24 @@ public ValidTxnList getValidTxns(long currentTxn) throws TException {
 
   @Override
   public ValidWriteIdList getValidWriteIds(String fullTableName) throws TException {
-    GetValidWriteIdsRequest rqst = new GetValidWriteIdsRequest(Collections.singletonList(fullTableName), null);
+    GetValidWriteIdsRequest rqst = new GetValidWriteIdsRequest(Collections.singletonList(fullTableName));
+    GetValidWriteIdsResponse validWriteIds = client.get_valid_write_ids(rqst);
+    return TxnCommonUtils.createValidReaderWriteIdList(validWriteIds.getTblValidWriteIds().get(0));
+  }
+
+  @Override
+  public ValidWriteIdList getValidWriteIds(String fullTableName, Long writeId) throws TException {
+    GetValidWriteIdsRequest rqst = new GetValidWriteIdsRequest(Collections.singletonList(fullTableName));
+    rqst.setWriteId(writeId);
```

Comment:
Set writeId only if writeId > 0.

Reply:
no need to put this restriction ..this function need not have any knowledge of valid/invalid write id